pkg/config/template.go

```diff
@@ -47,3 +64,74 @@ func getTemplates(templatePath string) (CiTemplates, error) {
 func isYAML(file string, info os.FileInfo) bool {
 	return !info.IsDir() && (filepath.Ext(file) == ".yaml" || filepath.Ext(file) == ".yml")
 }
+
+// TemplateCMController holds the details needed for the configmap controller
+type TemplateCMController struct {
+	cmclient  corev1.ConfigMapInterface
+	prNumber  int
+	logger    *logrus.Entry
+	templates CiTemplates
+}
+
+// NewTemplateCMController creates a new TemplateCMController
+func NewTemplateCMController(cmclient corev1.ConfigMapInterface, prNumber int, logger *logrus.Entry, templates CiTemplates) *TemplateCMController {
+	return &TemplateCMController{
+		cmclient:  cmclient,
+		prNumber:  prNumber,
+		logger:    logger,
+		templates: templates,
+	}
+}
+
+// CreateCMTemplates creates configMaps for all the changed templates.
+func (c *TemplateCMController) CreateCMTemplates() error {
+	var errors []error
+	s := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+
+	for filename, template := range c.templates {
+		buf := new(bytes.Buffer)
+
+		err := s.Encode(template, buf)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+		cmName := GetTempCMName(c.prNumber, template.Name)
+		cm := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: cmName,
+				Labels: map[string]string{
+					rehearseLabel: fmt.Sprintf("rehearse-temporary-cm-%d", c.prNumber),
```

Comment:
I would suggest re-using similar names to the upstream ones:

```yaml
  labels:
    created-by-prow: "true"
    prow.k8s.io/id: ff1f5cb0-4343-11e9-b256-0a58ac10d9a5
    prow.k8s.io/job: pull-ci-openshift-openshift-azure-master-e2e-azure-forceupdate
    prow.k8s.io/refs.org: openshift
    prow.k8s.io/refs.pull: "1261"
    prow.k8s.io/refs.repo: openshift-azure
    prow.k8s.io/type: presubmit
```

Reply:
Something like
```yaml
labels:
  created-by-pj-rehearse: "true"
  ci.openshift.org/rehearse/pull: "1234"
```